# Part -1 Python Basics (Variables)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [13]:
name = "Waleed"
father_name = "Waseem"
dob = "28-08-2005"

print(f"Name:\t\t{name}\nFather's Name:\t{father_name}\nDate of Birth:\t{dob}")

Name:		Waleed
Father's Name:	Waseem
Date of Birth:	28-08-2005


2. Write your small bio using variables and print it using print function

In [25]:
name = "Waleed"
father_name = "Waseem"
dob = "28-08-2005"
bio= f"My name is {name}\nfather's name is {father_name}\ndate of birth is {dob}"
print(bio)

My name is Waleed
father's name is Waseem
date of birth is 28-08-2005


3. Write a program in which use all the operators we can use in Python

In [37]:
a = 10
b = 3
print("Arithmetic Operators:")
print(f"Addition: {a} + {b} = {a + b}")
print(f"Subtraction: {a} - {b} = {a - b}")
print(f"Multiplication: {a} * {b} = {a * b}")
print(f"Division: {a} / {b} = {a / b}")
print(f"Floor Division: {a} // {b} = {a // b}")
print(f"Modulus: {a} % {b} = {a % b}")
print(f"Exponentiation: {a} ** {b} = {a ** b}")




print("\nComparison Operators:")
print(f"{a} == {b}: {a == b}")
print(f"{a} != {b}: {a != b}")
print(f"{a} > {b}: {a > b}")
print(f"{a} < {b}: {a < b}")
print(f"{a} >= {b}: {a >= b}")
print(f"{a} <= {b}: {a <= b}")



SyntaxError: invalid syntax (2549198783.py, line 1)

4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [ ]:
marks_english = 88
marks_islamiat = 90
marks_maths = 40

total_marks = 300

obtained_marks = marks_english + marks_islamiat + marks_maths
percentage = (obtained_marks / total_marks) * 100
print(f"Marks of English: {marks_english}")
print(f"Marks of Islamiat: {marks_islamiat}")
print(f"Marks of Maths: {marks_maths}")
print(f"Total Marks: {total_marks}")
print(f"Obtained Marks: {obtained_marks}")
print(f"Percentage: {percentage:.2f}%")

# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [31]:
yearofexperience =int(input("write your working experience "))
if(yearofexperience >5):
    salary= (int(input("what is your salary")))
    print("your bonus will be" ,0.05*salary)
else:
    print('you are not eligible for bonus')

write your working experience  2


you are not eligible for bonus



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [33]:
nationality = input("enter your nationality:")
if nationality == "pakistani":
    age = int(input("enter your age"))
    if age >= 18:
        print("you are eligible for vote")
    else:
        print("you are not eligible for vote")
else:
    print("you should be pakistani for vote")

enter your nationality: 29


you should be pakistani for vote



3) Write a program to check whether a number entered by user is even or odd.

In [8]:
number = int(input("Enter a number: "))

if number % 2 == 0:
    print(f"The number {number} is Even.")
else:
    print(f"The number {number} is Odd.")


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [47]:
number = int(input("Enter a number: "))

if number % 7 == 0:
    print(f"The number {number} is divisible.")
else:
    print(f"The number {number} is not divisible.")


Enter a number:  7


The number 7 is divisible.


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [39]:
number = int(input("Enter a number: "))

if number  % 5 == 0 :
    print("Hello.")
else:
    print("Bye.")

Enter a number:  88


Bye.


7) Write a program to display the last digit of a number.


In [49]:
number =int(input("Enter a number: "))
last_digit=abs(number) % 10
print(f"The last digit of the number is: {last_digit}")

Enter a number:  687


The last digit of the number is: 7


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [59]:
length = float(input("Enter the length of the rectangle: "))
breadth = float(input("Enter the breadth of the rectangle: "))

if length == breadth:
    print("The shape is a Square.")
else:
    print("The shape is a Rectangle.")

Enter the length of the rectangle:  5
Enter the breadth of the rectangle:  10


The shape is a Rectangle.


10) Take two int values from user and print greatest among them.

In [55]:
num1 =int(input("Enter a number"))
num2 =int(input("Enter a number"))
if num1 > num2:
    print(f"The greatest number is: {num1}")
elif num2 > num1:
    print(f"The greatest number is: {num2}")

Enter a number 78
Enter a number 25


The greatest number is: 78


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [57]:
quantity = int(input("Enter the quantity purchased: "))
unit_cost = 100


total_cost = quantity * unit_cost

if total_cost > 1000:
    discount = total_cost * 0.10 
    total_cost -= discount
    print(f"A 10% discount is applied.")


print(f"Total cost for the user is: {total_cost:.2f}")

Enter the quantity purchased:  67


A 10% discount is applied.
Total cost for the user is: 6030.00


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [29]:
marks =int(input("enter your marks"))
if marks>80:
    print("A") 
elif marks>=60 and marks<80:
    print("B")
elif marks>=50 and marks<60:
    print("C")
elif marks>=45 and marks<50:
    print("D")
elif marks>=25 and marks<45:
    print("E")
elif marks<=25 and marks>=0:
    print("F")
else:
    print("invalid error")

enter your marks 25


E


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [61]:
classes_held = int(input("Enter the number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))

attendance_percentage = (classes_attended / classes_held) * 100


print(f"Attendance percentage: {attendance_percentage:.2f}%")


if attendance_percentage >= 75:
    print("The student is allowed to sit in the exam.")
else:
    print("The student is not allowed to sit in the exam.")

Enter the number of classes held:  60
Enter the number of classes attended:  60


Attendance percentage: 100.00%
The student is allowed to sit in the exam.


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [63]:
classes_held = int(input("Enter the number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))
medical_issue =(input("Do you have any medical issue: "))

attendance_percentage = (classes_attended / classes_held) * 100


print(f"Attendance percentage: {attendance_percentage:.2f}%")


if attendance_percentage >= 75:
    print("The student is allowed to sit in the exam.")
else:
    print("The student is not allowed to sit in the exam.")

Enter the number of classes held:  6
Enter the number of classes attended:  9
Do you have any medical issue:  y


Attendance percentage: 150.00%
The student is allowed to sit in the exam.


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [71]:
year = int(input("Enter a year: "))


if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print(f"{year} is a Leap Year.")
else:
    print(f"{year} is not a Leap Year.")

Enter a year:  1000


1000 is not a Leap Year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [69]:
age = int(input("enter your age:"))
gender = input("enter your gender (M or F): :")
marriage = input("are you married (yes or no): :")
if gender == "F" :
    print("can only work in urban areas")
else:
    if age >=40 and age <=60:
        print("can work in urban areas only")
    elif age >=20 and age<40:
        print("can work anywhere")
    else:
        print("ERROR: you should be elegible you are under age or should get retire")

enter your age: 69
enter your gender (M or F): : y
are you married (yes or no): : yes


ERROR: you should be elegible you are under age or should get retire


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

In [73]:
units = int(input("Enter the number of units consumed: "))


if units <= 100:
    bill_amount = 0  
elif units <= 300:
    bill_amount = (units - 100) * 5 
else:
    bill_amount = (200 * 5) + ((units - 300) * 10) 


print(f"Total bill amount is Rs. {bill_amount}")

Enter the number of units consumed:  70


Total bill amount is Rs. 0


13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [65]:
age1=int(input('enter age1: '))
age2=int(input('enter age2: '))
age3=int(input('enter age3: '))
if age1>age2 and age1>age3:
    print('age1 is oldest')
elif age2>age1 and age2>age3:
    print('age2 is oldest')
else:
    print('age3 is oldest')
if age1<age2 and age1<age3:
    print('age1 is youngest')
elif age2<age1 and age2<age3:
    print('age2 is youngest')
else:
     print('age3 is youngest')


enter age1:  7
enter age2:  8
enter age3:  9


age3 is oldest
age1 is youngest
